### PR TITLE
Add normalization to Lie group constructor

### PR DESCRIPTION
**Pull request by fantaosha**
## Types of changes

- [ ] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have completed my CLA (see **CONTRIBUTING**)
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: theseus/geometry/so2.py [hunk 1]
```diff
@@ -121,15 +127,24 @@ def _project_impl(
             return torch.einsum("...k,...k", euclidean_grad, temp).unsqueeze(-1)
 
     @staticmethod
-    def _data_check(matrix: torch.Tensor):
-        if matrix.ndim != 2 or matrix.shape[1] != 2:
-            raise ValueError("2D rotations can only be 2D vectors.")
+    def _data_check_impl(matrix: torch.Tensor) -> bool:
+        with torch.no_grad():
+            if matrix.ndim != 2 or matrix.shape[1] != 2:
+                raise ValueError("2D rotations can only be 2D vectors.")
```
Comment (luisenp) [inline theseus/geometry/so2.py:133]: Reword as f"SO2 data tensors must be 2D tensors".

File: theseus/geometry/se3.py [hunk 2]
```diff
@@ -174,24 +176,24 @@ def _hat_matrix_check(matrix: torch.Tensor):
...
+                "The 3x3 top-left corner of hat matrices of SE3 can only be skew-symmetric."
             )
 
     @staticmethod
     def exp_map(
         tangent_vector: torch.Tensor, jacobians: Optional[List[torch.Tensor]] = None
     ) -> "SE3":
         if tangent_vector.ndim != 2 or tangent_vector.shape[1] != 6:
-            raise ValueError("Tangent vectors of SE(3) can only be 6-D vectors.")
+            raise ValueError("Tangent vectors of SE3 can only be 6-D vectors.")
```
Comment (luisenp) [inline theseus/geometry/se3.py:196]: Let's use 6D for consistency (no hyphen).

File: theseus/geometry/se2.py [hunk 3]
```diff
@@ -23,14 +23,14 @@ def __init__(
...
-        requires_check: bool = True,
+        strict: bool = False,
     ):
         if x_y_theta is not None and data is not None:
             raise ValueError("Please provide only one of x_y_theta or data.")
         if x_y_theta is not None:
             dtype = x_y_theta.dtype
-        if data is not None and requires_check:
-            self._data_check(data)
+        if data is not None:
```
Comment (luisenp) [inline theseus/geometry/se2.py:32]: Just realized that we can move this check (which happens in all subclasses) to `Manifold`'s constructor (right after line 40), so that the main logic is contained in one place, making it easier to maintain in the future. Sorry for missing this before!

**Review by luisenp — CHANGES_REQUESTED** [review]
Just a couple more changes, but almost there!

**Review by luisenp — APPROVED** [review]
Looks good to me, nice job!

**Review by mhmukadam — APPROVED** [review]
